.github/workflows/test-spec.yml

```diff
@@ -2,25 +2,32 @@ name: Test Spec
 
 on:
   push:
-    branches: ['**']
     paths:
-      - 'package*.json'
+      - .github/workflows/test-spec.yml
+      - package*.json
       - tsconfig.json
-      - 'tools/src/tester/**'
-      - 'spec/**'
+      - tools/src/tester/**'
+      - spec/**
   pull_request:
-    branches: ['**']
     paths:
-      - 'package*.json'
+      - .github/workflows/test-spec.yml
+      - package*.json
       - tsconfig.json
-      - 'tools/src/tester/**'
-      - 'spec/**'
+      - tools/src/tester/**
+      - spec/**
 
 jobs:
   test-opensearch-spec:
+    strategy:
+      matrix:
+        entry:
+          - {version: 2.15.0, hub: 'opensearchproject'}
+          - {version: 2.16.0, hub: 'opensearchstaging'}
```

Comment:
This is great. My concern is whether we should test multiple versions of OS now when the test framework cant differentiate which features in the spec can be tested against which version of OS. Say if we add a new query param in 2.17 and update the spec accordingly, then all matrix entries before 2.17 will likely fail stories that test the affected endpoint.

Potential solution (in a different PR): Add `min_cluster_version` and `max_cluster_version` properties for test stories and the test framework will only test the story if the it's being tested against a valid cluster version.

Reply:
I'm glad you asked, I got better in #410.